### PR TITLE
8309235: Unnamed Variables (_) can't be used in JShell

### PR DESCRIPTION
**Pull request by biboudis**
This PR addresses the issue of JShell not accepting unnamed local variables. This solution prints all local variables via the `/vars` command. e.g.,

```
jshell> int _ = 42;
 ==> 42

jshell> int _ = 43;
 ==> 43

jshell> String _ = "44";
 ==> "44"

jshell> /vars
|    int _ = 42
|    int _ = 43
|    String _ = "44"
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8309235](https://bugs.openjdk.org/browse/JDK-8309235): Unnamed Variables (_) can't be used in JShell (**Bug** - P4)

### Reviewers
 * [Adam Sotona](https://openjdk.org/census#asotona) (@asotona - **Reviewer**) ⚠️ Review applies to [953b5846](https://git.openjdk.org/jdk/pull/14337/files/953b5846692ead6944bde4122d8c5b9087123e27)

### Contributors
 * Jan Lahoda `<jlahoda@openjdk.org>`
 * Aggelos Biboudis `<abimpoudis@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14337/head:pull/14337` \
`$ git checkout pull/14337`

Update a local copy of the PR: \
`$ git checkout pull/14337` \
`$ git pull https://git.openjdk.org/jdk.git pull/14337/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14337`

View PR using the GUI difftool: \
`$ git pr show -t 14337`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14337.diff">https://git.openjdk.org/jdk/pull/14337.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14337#issuecomment-1578954440)

### CODE REVIEW

File: src/jdk.jshell/share/classes/jdk/internal/jshell/tool/JShellTool.java [hunk 1]
```diff
@@ -3350,7 +3350,8 @@ private boolean cmdVars(String arg) {
             String val = state.status(vk) == Status.VALID
                     ? feedback.truncateVarValue(state.varValue(vk))
                     : getResourceString("jshell.msg.vars.not.active");
-            hard("  %s %s = %s", vk.typeName(), vk.name(), val);
+            String varName = vk.name();
+            hard("  %s %s = %s", vk.typeName(),  varName.isEmpty() ? "_" : varName, val);
```
Comment (turbanoff) [inline src/jdk.jshell/share/classes/jdk/internal/jshell/tool/JShellTool.java:3354]:
```suggestion
            hard("  %s %s = %s", vk.typeName(), varName.isEmpty() ? "_" : varName, val);
```

**Review by asotona — APPROVED** [review]
Looks good 👍

**Pull request closed without merging**